### PR TITLE
feat(apps/prod/jenkins-beta): add a jenkins-beta for debug test

### PR DESCRIPTION
**Pull request by purelind**
Add a jenkins-beta for debug test.

### CODE REVIEW

File: apps/prod/jenkins-beta/pre/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  sourceRef:
+    kind: GitRepository
+    name: flux-system
+    namespace: flux-system
+  path: ./apps/prod/jenkins-beta/pre/secret
+  prune: true
+  postBuild:
+    substituteFrom:
+      - kind: Secret
+        name: ci-pipeline-cache
```
Comment (wuhuizuo) [inline apps/prod/jenkins-beta/pre/secret.yaml:21]:
```suggestion
        name: beta-ci-pipeline-cache
```

File: apps/prod/jenkins-beta/pre/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - kind: ConfigMap
+        name: ci-pipeline-cache
```
Comment (wuhuizuo) [inline apps/prod/jenkins-beta/pre/secret.yaml:23]:
```suggestion
        name: beta-ci-pipeline-cache
```

File: apps/prod/jenkins-beta/release/values-JCasC.yaml [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  # and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc')
+  additionalExistingSecrets:
+    # for git clone for private repo.
+    - { name: github, keyName: git-private-key }
+    - { name: github, keyName: git-username }
+    - { name: github, keyName: bot-token }
+    - { name: github, keyName: pr-diff-token }
+
+    # for ci cache, no need create manually.
+    - { name: jenkins-cache, keyName: region }
```
Comment (wuhuizuo) [inline apps/prod/jenkins-beta/release/values-JCasC.yaml:16]: it's "jenkins-beta-cache"

File: apps/prod/jenkins-beta/release/values-controller.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+controller:
+  image: "hub.pingcap.net/jenkins/jenkins-with-plugins"
+  tag: "2.387.3-jdk11"
+  resources:
+    requests:
+      cpu: "16"
+      memory: 80Gi
+    limits:
+      cpu: "16"
+      memory: 80Gi
```
Comment (wuhuizuo) [inline apps/prod/jenkins-beta/release/values-controller.yaml:11]: the bot found the issue.

File: apps/prod/jenkins-beta/release/values-controller.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+  # jenkinsOpts: ""
+  # If you are using the ingress definitions provided by this chart via the `controller.ingress` block the configured hostname will be the ingress hostname starting with `https://` or `http://` depending on the `tls` configuration.
+  # The Protocol can be overwritten by specifying `controller.jenkinsUrlProtocol`.
+  # jenkinsUrlProtocol: "https"
+  # If you are not using the provided ingress you can specify `controller.jenkinsUrl` to change the url definition.
+  # jenkinsUrl: ""
+  # If you set this prefix and use ingress controller then you might want to set the ingress path below
+  jenkinsUriPrefix: /jenkins
+  jenkinsUrl: https://internal.do.pingcap.net/jenkins
```
Comment (wuhuizuo) [inline apps/prod/jenkins-beta/release/values-controller.yaml:34]:
```suggestion
  jenkinsUrl: http://internal.do.pingcap.net:30080/jenkins
```

File: apps/prod/jenkins-beta/release/values-controller.yaml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  jenkinsAdminEmail: lifu.wu@pingcap.net
+
+  ingress:
+    enabled: true
```
Comment (wuhuizuo) [inline apps/prod/jenkins-beta/release/values-controller.yaml:38]: Do we need the ingress?

Comment (purelind) [reply apps/prod/jenkins-beta/release/values-controller.yaml:38]: Currently, it is needed.

**Comment by purelind** [comment]
> ```
> 3. Test the newly added `jenkins-beta` in a separate namespace or test environment to avoid any interference with existing services and resources.
> ```
 3. Test the newly added `jenkins-beta` in a separate namespace or test environment to avoid any interference with existing services and resources.
 
Is suggestion 3 reasonable? Can it be deployed to another namespace?

File: apps/prod/jenkins-beta/release/values-service-account.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+serviceAccount:
+  create: true
+  imagePullSecretName: jenkins-registry-cred
```
Comment (wuhuizuo) [inline apps/prod/jenkins-beta/release/values-service-account.yaml:3]: should be another name.

**Review by wuhuizuo — APPROVED** [review]

**Pull request merged**